client/state/data-layer/wpcom/sites/admin-menu/index.js

```diff
@@ -19,12 +19,15 @@ export const requestFetchAdminMenu = ( action ) =>
 		action
 	);
 
-const sanitizeUrl = ( url, wpAdminUrl ) => {
+const sanitizeUrl = ( url, wpAdminUrl, siteUrl ) => {
 	const isSafeInternalUrl = new RegExp( '^/' ).test( url );
 	// The replace function removes the protocol.
 	const isSafeWpAdminUrl = new RegExp( `^${ wpAdminUrl?.replace( /^https?:\/\//, '' ) }` ).test(
 		url?.replace( /^https?:\/\//, '' )
 	);
+	const isSafeJetpackRedirectUrl =
+		new RegExp( '^https://jetpack.com/redirect/' ) &&
+		url?.includes( `site=${ siteUrl.replace( /^https?:\/\//, '' ) }` );
 
```

Comment:
We consider a URL safe if:
1. It is Jetpack Redirect URL (starts with `https://jetpack.com/redirect/`).
2. Contains the URL of the current site in the `site` query parameter.